### DISCUSSION TITLE
Socket connection breaks after inactivity for 12-15 secs

### DISCUSSION THREAD

**Discussion by shrimad-mishra-cognoai** [deepgram/community#419] [General help]
I am using deepgram for real-time transcription, but the socket connection breaks in between. On the doc page, I saw Keep Alive Heartbeat, but it is not supported in the current Python SDK version, can anyone help here?

**Comment by jkroll-deepgram** [comment]
Hi @shrimad-mishra-cognoai, thank you for raising this point. We'd merged a [PR](https://github.com/deepgram/deepgram-python-sdk/pull/120) to add KeepAlive to the Python SDK, but did not subsequently do a version release to make it easily available.

If you upgrade to the latest Python SDK version, [2.12.0](https://github.com/deepgram/deepgram-python-sdk/releases/tag/v2.12.0), then you can use the new `.keep_alive()` method in this pattern:
```
deepgram = Deepgram(DEEPGRAM_API_KEY)
deepgramLive = await deepgram.transcription.live(PARAMS)
deepgramLive.keep_alive()
```

Per our [docs](https://developers.deepgram.com/reference/streaming#stream-keepalive), the streaming connection is intended to close after 12 seconds without audio, so you can periodically send a KeepAlive message for as long as you'd like to maintain the open connection.
